shipyard-controller/nats/connection.go

```diff
@@ -41,13 +41,12 @@ type NatsConnectionHandler struct {
 	subscriptions  []*PullSubscription
 	topics         []string
 	natsURL        string
-	messageHandler IKeptnNatsMessageHandler
 	ctx            context.Context
 	jetStream      nats.JetStreamContext
 }
 
-func NewNatsConnectionHandler(ctx context.Context, natsURL string, messageHandler IKeptnNatsMessageHandler) *NatsConnectionHandler {
-	return &NatsConnectionHandler{natsURL: natsURL, messageHandler: messageHandler, ctx: ctx}
+func NewNatsConnectionHandler(ctx context.Context, natsURL string) *NatsConnectionHandler {
+	return &NatsConnectionHandler{natsURL: natsURL, ctx: ctx}
```

Comment:
the natsConnectionHandler is now also responsible for providing a publisher, so the topics should be defined later when activating the subscription